connectors/src/connectors/google_drive/index.ts

```diff
@@ -539,14 +541,22 @@ export class GoogleDriveConnectorManager extends BaseConnectorManager<null> {
       }
     }
 
-    if (shouldFullSync) {
+    if (addedFolderIds.length > 0) {
       const res = await launchGoogleDriveFullSyncWorkflow(
         this.connectorId,
-        null
+        null,
+        addedFolderIds
       );
       if (res.isErr()) {
         return res;
       }
+    } else if (removedFolderIds.length > 0) {
+      // If we have added folders, the garbage collector will be automatically at the end of the full sync,
+      // but if we only removed folders, we need launch it manually.
```

Comment:
Not sure I understand -- isn't the garbage collector always running for all gdrive connectors ?

Reply:
See this:
https://github.com/dust-tt/dust/pull/6711#issuecomment-2277356269

My understanding is that, since we are no longer running the full sync if the delta is only removed folders, then we will not have a garbage collector happening.

Reply:
Then I guess we need to also ensure that `addedFolderIds.length === 0` given that we can add and remove folders in the same operation.

Also here garbage collect is going to be heavy and browse the whole files already synced, as it defaults to `Date.now()` for `lastSeen`. So basically all docs with a `lastSeen` older than now will be verified. I do think that a proper workflow just to remove some files of a known folders might be better.

Reply:
Note that it will introduce tricky edge cases, not sure it's worth it asap, i lack context on performance)

Reply:
Okok fine by me.

Reply:
>If I understand your comment correctly: it's an else if, the first if is addedFolderIds.length > 0 so we cannot reach that line unless addedFolderIds.length == 0.

Ah yeah I missed it.